docs/source/format/CanonicalExtensions.rst

```diff
@@ -148,6 +148,112 @@ Fixed shape tensor
   by this specification. Instead, this extension type lets one use fixed shape tensors
   as elements in a field of a RecordBatch or a Table.
 
+.. _variable_shape_tensor_extension:
+
+Variable shape tensor
+=====================
+
+* Extension name: `arrow.variable_shape_tensor`.
+
+* The storage type of the extension is: ``StructArray`` where struct
+  is composed of **data** and **shape** fields describing a single
+  tensor per row:
+
+  * **data** is a ``List`` holding tensor elements of a single tensor.
+    Data type of the list elements is uniform across the entire column.
+  * **shape** is a ``FixedSizeList<uint32>[ndim]`` of the tensor shape where
```

Comment:
I have not followed this, but was there any discussion of signed vs. unsigned here? Most of the Arrow spec uses signed integers even for numbers that are always positive (and unless your tensors are huge I'm not sure having one less bit to represent the shape's dimensions is a problem?).

Reply:
I don't think this was discussed and indeed `int64` would be more consistent, will change.

Reply:
I think `int32` is also fine (just not `uint32`)? (we have other places where we used int32 by default for positive integers, eg the offsets)